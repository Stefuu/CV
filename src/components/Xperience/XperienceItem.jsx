import React from 'react'
import styled from 'styled-components'
import Box from '../elements/Box'

const Ul = styled.ul`
  list-style: none;
  margin: 0;
  padding: 0;
`

const XperienceItem = (props) => (
  <Box
    lineHeight='1.38'
    width='100%'
    mb={20}>
    <Box
      mb={5}
      fontSize={18}>
      <strong>{props.company}</strong>
    </Box>
    <Box ml={10}>
      <Box
        mb={10}
        fontSize={16}>
        <span>{props.duration}</span>
      </Box>
      <Box mb={10}>
        <span>{props.position}</span>
      </Box>
      <Box
        width='90%'
        mb={10}>
        <span>{props.description}</span>
      </Box>
      <Box
        width='90%'>
        <Box><span>Destaques:</span></Box>
        {
          props.highlights &&
            <Ul>
              {props.highlights.map((item, index) => (
                <Box
                  key={index}
                  as='li'
                  mt={index > 0 ? 5 : 0}
                >{item}</Box>
              ))}
            </Ul>
        }
      </Box>
    </Box>
  </Box>
)

export default XperienceItem
