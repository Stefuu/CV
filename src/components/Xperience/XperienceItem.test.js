import React from 'react'
import { render } from 'react-testing-library'
import XperienceItem from './XperienceItem'

const data = {
  'company': '- KAITS - Sistema educacional',
  'duration': 'De: Fev/2011 até Jan/2012',
  'position': 'Cargo: Estagiário',
  'description': 'Descrição: Atuação com javascript vanilla na construção e manutenção de um sistema web para escolas de idioma',
  'highlights': [
    '-Uso de javascript sem nenhum framework foi uma experiência interessante, principalmente por estar começando a usar js na época'
  ]
}

describe('XperienceItem', () => {
  it('Deve renderizar [snapshot]', async () => {
    const { container } = render(<XperienceItem {...data} />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
