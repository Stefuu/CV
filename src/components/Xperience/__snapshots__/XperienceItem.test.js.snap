// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`XperienceItem Deve renderizar [snapshot] 1`] = `
<div
  class="sc-bdVaJa UqLRs"
>
  <div
    class="sc-bdVaJa cqSdRU"
  >
    <strong>
      - KAITS - Sistema educacional
    </strong>
  </div>
  <div
    class="sc-bdVaJa hOejgn"
  >
    <div
      class="sc-bdVaJa cZHmfT"
    >
      <span>
        De: Fev/2011 até Jan/2012
      </span>
    </div>
    <div
      class="sc-bdVaJa nLDLh"
    >
      <span>
        Cargo: Estagiário
      </span>
    </div>
    <div
      class="sc-bdVaJa iSxgmk"
    >
      <span>
        Descrição: Atuação com javascript vanilla na construção e manutenção de um sistema web para escolas de idioma
      </span>
    </div>
    <div
      class="sc-bdVaJa eCYppH"
    >
      <div
        class="sc-bdVaJa hvxHyA"
      >
        <span>
          Destaques:
        </span>
      </div>
      <ul
        class="sc-bwzfXH bJZRKI"
      >
        <li
          class="sc-bdVaJa cWwZGa"
        >
          -Uso de javascript sem nenhum framework foi uma experiência interessante, principalmente por estar começando a usar js na época
        </li>
      </ul>
    </div>
  </div>
</div>
`;
